griptape/structures/pipeline.py

```diff
@@ -71,10 +71,7 @@ def context(self, task: BaseTask) -> dict[str, Any]:
         return context
 
     def __run_from_task(self, task: Optional[BaseTask]) -> None:
-        if task is None:
+        if task is None or isinstance(task.execute(), ErrorArtifact) and self.fail_fast:
```

Comment:
I know `and` has a higher precedence than `or`, so this will evaluate correctly, but might be nice to add parenthesis to make it a little faster to parse (not everyone remembers operator precedence).